CHANGELOG.md

```diff
@@ -1,3 +1,436 @@
+    ([#657](https://github.com/kinvolk/lokomotive/pull/657)).
+
+- Improve documentation on using remote
+    [backends](https://github.com/kinvolk/lokomotive/blob/v0.3.0/docs/configuration-reference/backend/s3.md)
+    ([#670](https://github.com/kinvolk/lokomotive/pull/670)).
+
+- How to guide for setting up monitoring on Lokomotive
+    ([#480](https://github.com/kinvolk/lokomotive/pull/480)).
+
+- Add `codespell` section in [development
+    documentation](https://github.com/kinvolk/lokomotive/blob/v0.3.0/docs/development/README.md)
+    ([#700](https://github.com/kinvolk/lokomotive/pull/700)).
+
+- Include a demo GIF in the readme
+    ([#636](https://github.com/kinvolk/lokomotive/pull/636)).
+
+#### Bugfixes
+
+- Remove contour ingress workaround (due to an [upstream issue](https://github.com/projectcontour/contour/issues/403))
+  for ExternalDNS ([#635](https://github.com/kinvolk/lokomotive/pull/635)).
+
+#### Development
+
+- Disable marking releases as `pre-releases` ([#678](https://github.com/kinvolk/lokomotive/pull/678)).
+
+- Do not show release values in terraform output
+    ([#627](https://github.com/kinvolk/lokomotive/pull/627)).
+
+- Change NewClientset to create Kubernetes Clientset from kubeconfig file
+content, instead of from file path
+([#628](https://github.com/kinvolk/lokomotive/pull/628)).
+
+- Remove provider aliases from go templates
+    ([#617](https://github.com/kinvolk/lokomotive/pull/617)).
+
+
+#### Miscellaneous
+
+- Following flatcar-linux/Flatcar#123, Flatcar 2513.1.0 for ARM contains the dig
+    binary so the workaround is no longer needed
+    ([#703](https://github.com/kinvolk/lokomotive/pull/703)).
+
+- Improve error message for `wait-for-dns` output
+    ([#735](https://github.com/kinvolk/lokomotive/pull/735)).
+
+- Add `codespell` to enable spell check on all PRs ([#661](https://github.com/kinvolk/lokomotive/pull/661)).
+
+### Upgrading from v0.2.1
+
+#### Configuration syntax changes
+
+There have been some minor changes to the configurations of following components:
+* contour
+* prometheus-operator.
+
+Please make sure new the configuration structure is in place before the upgrade.
+
+##### Contour component
+
+Optional `ingress_hosts` variable is now removed.
+
+old:
+
+```hcl
+component "contour" {
+  .
+  .
+  ingress_hosts = ["*.example.lokomotive-k8s.net"]
+}
+```
+
+new:
+
+```hcl
+component "contour" {
+  .
+  .
+}
+
+```
+
+##### Prometheus-operator component
+
+* Prometheus specific variables are now under a `prometheus` block.
+* Variable `external_url` is now removed and now configured under
+    `prometheus.ingress.host`. Remove the http scheme when configuring.
+* A new optional `prometheus.ingress` sub-block is introduced to expose
+Prometheus over ingress.
+
+old:
+
+```hcl
+component "prometheus-operator" {
+  .
+  .
+  prometheus_metrics_retention = "14d"
+  prometheus_external_url      = "https://api.example.com/prometheus"
```

Comment:
We should use the same value for old and new config version, so user can easily map them, as there is no new values introduced here.
```suggestion
  prometheus_external_url      = "https://prometheus.example.lokomotive-k8s.net"
```